bundles/org.openhab.binding.somfymylink/src/main/java/org/openhab/binding/somfymylink/internal/discovery/SomfyMyLinkDeviceDiscoveryService.java

```diff
@@ -0,0 +1,191 @@
+        logger.debug("Starting Somfy My Link background discovery");
+
+        ScheduledFuture<?> discoveryJob = this.discoveryJob;
+        if (discoveryJob == null || discoveryJob.isCancelled()) {
+            discoveryJob = scheduler.scheduleWithFixedDelay(this::discoverDevices, 10, DISCOVERY_REFRESH_SEC,
+                    TimeUnit.SECONDS);
+        }
+    }
+
+    @Override
+    protected void stopBackgroundDiscovery() {
+        logger.debug("Stopping Somfy MyLink background discovery");
+
+        ScheduledFuture<?> discoveryJob = this.discoveryJob;
+        if (discoveryJob != null) {
+            discoveryJob.cancel(true);
+            this.discoveryJob = null;
+        }
+    }
+
+    @Override
+    protected synchronized void startScan() {
+        Future<?> scanTask = this.scanTask;
+        if (scanTask == null || scanTask.isDone()) {
+            logger.debug("Starting somfy mylink discovery scan");
+            scanTask = scheduler.submit(this::discoverDevices);
+        }
+    }
+
+    @Override
+    public void stopScan() {
+        Future<?> scanTask = this.scanTask;
+        if (scanTask != null) {
+            logger.debug("Stopping somfy mylink discovery scan");
+            scanTask.cancel(true);
+        }
+        super.stopScan();
+    }
+
+    private synchronized void discoverDevices() {
+        logger.info("Scanning for things...");
+
+        if (this.mylinkHandler.getThing().getStatus() != ThingStatus.ONLINE) {
+            logger.info("Skipping device discover as bridge is {}", this.mylinkHandler.getThing().getStatus());
+            return;
+        }
+
+        try {
+            // get the shade list
+            SomfyMyLinkShade[] shades = this.mylinkHandler.getShadeList();
+
+            for (SomfyMyLinkShade shade : shades) {
+                String id = shade.getTargetID();
+                String label = "Somfy Shade " + shade.getName();
+
+                if (id != null) {
+                    logger.debug("Adding device {}", id);
+                    notifyThingDiscovery(THING_TYPE_SHADE, id, label, TARGET_ID);
+                }
+            }
+
+            SomfyMyLinkScene[] scenes = this.mylinkHandler.getSceneList();
+
+            for (SomfyMyLinkScene scene : scenes) {
+                String id = scene.getTargetID();
+                String label = "Somfy Scene " + scene.getName();
+
+                logger.debug("Adding device {}", id);
+                notifyThingDiscovery(THING_TYPE_SCENE, id, label, SCENE_ID);
+            }
+        } catch (SomfyMyLinkException e) {
+            logger.warn("Error scanning for devices: {}", e.getMessage(), e);
+            ScanListener scanListener = this.scanListener;
+            if (scanListener != null) {
+                scanListener.onErrorOccurred(e);
+            }
+        }
+    }
+
+    private void notifyThingDiscovery(ThingTypeUID thingTypeUID, String id, String label, String idType) {
+        if (StringUtils.isEmpty(id)) {
```

Comment:
We're working on removing the apache commons dependencies. Therefor can you change it to: 
```suggestion
        if (id ==null || id.isEmpty()) {
```

Reply:
done.